packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -831,7 +833,7 @@ class ReactDOMServerRenderer {
     parentNamespace: string,
   ): string {
     if (typeof child === 'string' || typeof child === 'number') {
-      const text = '' + child;
+      const text = toString(getToStringValue(child));
```

Comment:
I don't like how both of these functions are essential to safely cast values to strings. I'd like to also export a function that just calls both of these.

Reply:
We should probably get rid of this interim value completely. I’ve added it because we used this pattern in the input code but I feel like it caused a lot of confusion already 😕 

Maybe we can keep the original value as `mixed` in the wrapper state? 

I can work on that next week.